hw8-AVLtree/AVLtree.c

```diff
@@ -0,0 +1,436 @@
+﻿#define _CRT_SECURE_NO_WARNINGS
+
+#include <string.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+typedef struct Node
+{
+    struct Node* leftSon;
+    struct Node* rightSon;
+    struct Node* parent;
+    int key;
+    char* value;
+    int balance;
+} Node;
+
+typedef struct Tree
+{
+    Node* root;
+}
+Tree;
+
+Tree* createTree()
+{
+    return calloc(sizeof(Tree), 1);
+}
+
+bool isEmpty(Tree* tree)
+{
+    return tree->root == NULL;
+}
+
+Node* leftRotation(Node* root)
+{
+    Node* base = root->rightSon;
+    root->rightSon = base->leftSon;
+    if (base->leftSon)
```

Comment:
Лучше 
```suggestion
    if (base->leftSon != NULL)
```
Чтобы явно написать, что речь про указатели и их равенство NULLу, не полагаясь на автоматическое приведение типов